example/src/main/java/com/stripe/example/view/CreditCardView.java

```diff
@@ -0,0 +1,560 @@
+                    freeze = false;
+
+                    boolean error = false;
+
+                    if (!StripeTextUtils.isBlank(monthStr) || !StripeTextUtils.isBlank(yearStr)) {
+                        if ((!StripeTextUtils.isBlank(monthStr) && monthStr.length() == 2 && !mCard.validateExpMonth())) {
+                            mEtExpiryDate.setTextColor(ContextCompat.getColor(getContext(), android.R.color.holo_red_dark));
+                            if (mCallback != null) {
+                                mCallback.onError(ERROR_EXPIRY_MONTH);
+                            }
+                            error = true;
+                        } else if ((!StripeTextUtils.isBlank(yearStr) && yearStr.length() == 2 && !mCard.validateExpYear())) {
+                            mEtExpiryDate.setTextColor(ContextCompat.getColor(getContext(), android.R.color.holo_red_dark));
+                            if (mCallback != null) {
+                                mCallback.onError(ERROR_EXPIRY_YEAR);
+                            }
+                            error = true;
+                        } else if (!StripeTextUtils.isBlank(yearStr) && yearStr.length() == 2) { // only move to next field when year entered
+                            mEtCvcNum.requestFocus();
+                        }
+                    }
+                    if (!error) {
+                        if (mCallback != null) {
+                            mCallback.onClearError();
+                        }
+                    }
+
+                    sendCardValidationCallback();
+                }
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                //not used
+            }
+        });
+
+        mEtCvcNum.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View view, boolean focus) {
+                if (focus) {
+                    setCreditCardIconForCvc();
+                } else {
+                    setCreditCardIconForNumber();
+                }
+            }
+        });
+
+        mEtCvcNum.addTextChangedListener(new TextWatcher() {
+            boolean freeze = false;
+            int beforeStringLen = 0;
+
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int start, int count, int after) {
+                if (!freeze) {
+                    beforeStringLen = mEtCvcNum.getText().toString().length();
+                }
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int start, int before, int count) {
+                if (!freeze) {
+                    String curString = mEtCvcNum.getText().toString();
+                    int curStringLen = curString.length();
+
+                    // TODO : this is deprecated, but what else should be used?
+                    mCard.setCVC(curString);
```

Comment:
Understood. I think that the setters are the simplest way to validate as we go, it's piecemeal for expiry date and CVC too, but could replace with Card.Builder if a requirement. Let me know if you want me to go ahead with that.